ide/svg/src/org/netbeans/modules/svg/navigation/SvgNavigatorPanel.java

```diff
@@ -0,0 +1,209 @@
+     */
+    private Lookup.Result<SVGDataObject> currentContext;
+    /**
+     * listener to context changes
+     */
+    private LookupListener contextListener;
+    /**
+     * Listens for changes on image file.
+     */
+    private FileChangeListener fileChangeListener;
+    private long lastSaveTime = -1;
+    private DataObject currentDataObject;
+    private static final RequestProcessor WORKER = new RequestProcessor(SvgNavigatorPanel.class.getName());
+
+    @Override
+    public String getDisplayName() {
+        return NbBundle.getMessage(SvgNavigatorPanel.class, "Navigator_DisplayName");
+    }
+
+    @Override
+    public String getDisplayHint() {
+        return NbBundle.getMessage(SvgNavigatorPanel.class, "Navigator_DisplayHint");
+    }
+
+    @Override
+    public JComponent getComponent() {
+        if (lastSaveTime == -1) {
+            lastSaveTime = System.currentTimeMillis();
+        }
+        if (panelUI == null) {
+            panelUI = new SvgPreviewPanel();
+        }
+        return panelUI;
+    }
+
+    @Override
+    public void panelActivated(Lookup context) {
+        // lookup context and listen to result to get notified about context changes
+        currentContext = context.lookup(MY_DATA);
+        currentContext.addLookupListener(getContextListener());
+        // get actual data and recompute content
+        Collection<? extends SVGDataObject> data = currentContext.allInstances();
+        currentDataObject = getDataObject(data);
+
+        if (currentDataObject == null) {
+            return;
+        }
+
+        if (fileChangeListener == null) {
+            fileChangeListener = new SvgFileChangeAdapter();
+        }
+        currentDataObject.getPrimaryFile().addFileChangeListener(fileChangeListener);
+        setNewContent(currentDataObject);
+    }
+
+    @Override
+    public void panelDeactivated() {
+        currentContext.removeLookupListener(getContextListener());
+        currentContext = null;
+        if (currentDataObject != null) {
+            currentDataObject.getPrimaryFile().removeFileChangeListener(fileChangeListener);
+        }
+        currentDataObject = null;
+    }
+
+    @Override
+    public Lookup getLookup() {
+        // go with default activated Node strategy
+        return null;
+    }
+
+    private void setNewContent(final DataObject dataObject) {
+        if (dataObject == null) {
+            return;
+        }
+
+        WORKER.post(() -> {
+            FileObject fo = dataObject.getPrimaryFile();
+
+            if (fo == null) {
+                return;
+            }
+
+            if (panelUI == null) {
+                return;
+            }
+
+            try {
+                SVGLoader svgLoader = new SVGLoader();
+                SVGDocument svgDocument = svgLoader.load(fo.toURL());
+
+                SwingUtilities.invokeLater(() -> panelUI.setSvg(svgDocument));
+            } catch (Exception ex) {
+                LOG.log(Level.SEVERE, ex.getMessage());
```

Comment:
```java
                LOG.log(Level.SEVERE, "can not load svg: " + fo, ex);
```
retains stack and lists the file. Note that `SVGViewerElement` does also load SVGs, see other comment.

Reply:
try catch can be removed since loading does not throw exceptions, see https://github.com/apache/netbeans/pull/7938#discussion_r1893716745